supreme/src/iosMain/kotlin/at/asitplus/signum/supreme/sign/PrivateKeySignerImpl.kt

```diff
@@ -1,68 +1,18 @@
 package at.asitplus.signum.supreme.sign
 
-import at.asitplus.KmmResult
 import at.asitplus.signum.indispensable.*
-import at.asitplus.signum.internals.*
-import at.asitplus.signum.supreme.*
 import kotlinx.cinterop.ExperimentalForeignApi
-import platform.Foundation.NSData
-import platform.Security.SecKeyCreateSignature
-import platform.Security.SecKeyRef
-
 
 @OptIn(ExperimentalForeignApi::class)
 actual fun makePrivateKeySigner(
     key: CryptoPrivateKey.RSA,
     algorithm: SignatureAlgorithm.RSA
-): Signer.RSA = RSAPrivateKeySigner(key, algorithm, key.publicKey)
+): Signer.RSA =
+    key.toSecKey().mapCatching { EphemeralSigner.RSA(EphemeralSignerConfiguration(), it, key.publicKey, algorithm) }.getOrThrow()
```

Comment:
remove `PrivateKeySigner` which was just `EphemeralKeySigner` except it leaks memory